### PR TITLE
New gst lead function done

### PR DESCRIPTION
**Pull request by AshishPis3**

### CODE REVIEW

File: functions/src/leads/newleads.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        const crm_data = await admin.firestore().collection('token').doc('crm').get();
+        if(crm_data.data() === undefined){
+            throw Error('Invalid Token');
+        }
+        request({
+            headers: {
+              'Content-Type': 'application/json',
+              'authorization': `${crm_data.data().access_token}`
+            },
+            uri: 'https://crm.indiacorporates.com/Api//V8/module',
```
Comment (ankitpise) [inline functions/src/leads/newleads.ts:32]: Use config variable for https://crm.indiacorporates.com

File: functions/src/leads/newleads.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            body: JSON.stringify(mainData),
+            method: 'POST'
+          }, (err, response, body) => {
+            if (err) {
+                res.send({Error : err});
+            }else {
+                res.send(body);
```
Comment (ankitpise) [inline functions/src/leads/newleads.ts:39]: @mapuskar did you manage to hide unnecessary fields with the response?
@AshishPis3 what data from response we need for further operations exactly?

File: functions/src/leads/newleads.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            }
+          });
+    })
+})
+
+const validator = require('../lib/validator');
+
+function leadDataValidator(body){
```
Comment (ankitpise) [inline functions/src/leads/newleads.ts:47]: why this function is not in validator library? It seems like its validating the data

File: functions/src/lib/validator.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+export const newGstLeadValidator = function (body) {
+    if (body.first_name === undefined || body.first_name.length < 4) {
```
Comment (ankitpise) [inline functions/src/lib/validator.ts:2]: @AshishPis3 checkout https://www.npmjs.com/package/parse-full-name library.

This returns proper data with salutation, firstname, middle name, last name etc.

If you find it useful use it.

**Review by ankitpise — CHANGES_REQUESTED** [review]

File: functions/src/leads/newleads.ts (hunk 2 above)
Comment (AshishPis3) [reply functions/src/leads/newleads.ts:39]: Lead Id and other message ok

File: functions/src/leads/newleads.ts (hunk 3 above)
Comment (AshishPis3) [reply functions/src/leads/newleads.ts:47]: It will check lead type before calling specific lead type validation

**Review by ankitpise — APPROVED** [review]

**Review by ankitpise — APPROVED** [review]

**Pull request merged**